src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -451,86 +567,91 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
-     * matching itinerary isn't found on the next possible date the trip should be monitored on.
-     */
-    private boolean calculateNextItinerary() {
-        // double check that this trip is in fact active and throw an error if it is not
-        if (trip.isInactive()) {
-            throw new RuntimeException("An attempt was made to find the next possible active date of an inactive monitored trip.");
-        }
-
-        // if the trip is not active on the current zoned date time, advance until a day is found when the trip is
-        // active.
+    private void advanceToNextActiveTripDate() throws URISyntaxException {
+        // Advance the target date/time until a day is found when the trip is active.
         while (!trip.isActiveOnDate(targetZonedDateTime)) {
             targetZonedDateTime = targetZonedDateTime.plusDays(1);
         }
         targetDate = targetZonedDateTime.format(DateTimeUtils.DEFAULT_DATE_FORMATTER);
 
-        // execute trip plan request for the target time
-        LOG.info("Attempting to calculate next trip on target date {}", targetDate);
-        if (!calculateOtpResponse()) {
-            // failed to calculate, return false
-            return false;
+        LOG.info("Next itinerary happening on {}.", targetDate);
+
+        // Update the matching itinerary with the expected scheduled times for when the next trip is
+        // expected to happen. This is done by incrementing the matching itinerary's start or end time such that it
+        // would be the either the first possible itinerary happening after the target time for depart at trips or
+        // the latest possible itinerary that ends before the target time for arrive by trips. Doing so this way
+        // avoids making calls to OTP and sets up an expected itinerary to check for even if OTP is unable to return
+        // the itinerary due to it being suboptimal.
+        long offsetMillis;
+        if (trip.isArriveBy()) {
+            // find the closest itinerary end time that does not exceed the target zoned date time
+            ZonedDateTime newEndTime = DateTimeUtils.makeOtpZonedDateTime(matchingItinerary.endTime);
+            do {
+                newEndTime = newEndTime.plusDays(1);
+            } while (newEndTime.plusDays(1).isBefore(targetZonedDateTime));
+            offsetMillis = newEndTime.toInstant().toEpochMilli() - matchingItinerary.endTime.getTime();
+        } else {
+            // find the first itinerary start time that does exceeds the target zoned date time
+            ZonedDateTime newStartTime = DateTimeUtils.makeOtpZonedDateTime(matchingItinerary.startTime);
+            do {
+                newStartTime = newStartTime.plusDays(1);
+            } while (newStartTime.isBefore(targetZonedDateTime));
+            offsetMillis = newStartTime.toInstant().toEpochMilli() - matchingItinerary.startTime.getTime();
```

Comment:
I think this code block could be placed in a new method to improve readability of code.

Reply:
I agree that the `Itinerary#offsetTimes` method is useful and implemented it in 1987213b6754ffa98f4748820b628be64f9443f4. However, creating a new method for `getNextTripOffsetMillis` doesn't seem quite worth it as that logic is quite central to the method it is already in, so I think it'd be better for it to stay in this method.